lib/justifi/list_object.rb

```diff
@@ -1,11 +1,28 @@
 # frozen_string_literal: true
 
 module Justifi
-  class ListObject < JustifiObject
+  class ListObject
+    extend Forwardable
+
+    attr_reader :justifi_object
+    attr_accessor :request_params, :request_headers, :path
+
+    def_delegators :@justifi_object, :data, :data
+    def_delegators :@justifi_object, :page_info, :page_info
+    def_delegators :@justifi_object, :http_status, :http_status
+    def_delegators :@justifi_object, :raw_response, :raw_response
+
```

Comment:
Went from an inheritance approach to a wrapper structure. I tried not to touch all the methods that were working and added changed only the `Justifi::ListObject.list`  method

Reply:
So ended up delegating main methods to the specific justifi_object